service/topology_coordinator.cc

```diff
@@ -752,6 +753,87 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
+                                repl_strategy, table, tmptr, new_rf_per_dc_map);
+                    } catch(const std::exception& e) {
+                        new_data.status = tablet_replica_calculation_status::other;
+                        error = e.what();
+                        rtlogger.error("{}", error);
+                    }
+
+                    if (new_data.status != tablet_replica_calculation_status::success) {
+                        rtlogger.error("Couldn't process global_topology_request::keyspace_rf_change, status: {}, rf per dc map: {}",
+                                        new_data.status, new_data.new_tablet_map);
+                        updates.clear(); // remove all tablets mutations ...
+                        break;           // ... and only create mutations deleting the global req
+                    }
+
+                    auto tablet_id = new_data.new_tablet_map.first_tablet();
+                    while (true) {
+                        auto& tablet_info = new_data.new_tablet_map.get_tablet_info(tablet_id);
+                        auto last_token = new_data.new_tablet_map.get_last_token(tablet_id);
+                        updates.emplace_back(replica::tablet_mutation_builder(guard.write_timestamp(), table->id())
+                                                 .set_new_replicas(last_token, tablet_info.replicas)
+                                                 .set_stage(last_token, locator::tablet_transition_stage::allow_write_both_read_old)
+                                                 .set_transition(last_token, locator::tablet_transition_kind::rf_change)
+                                                 .build());
+
+                        if (auto next_tablet = new_data.new_tablet_map.next_tablet(tablet_id); next_tablet.has_value()) {
+                            tablet_id = *next_tablet;
+                        } else {
+                            break;
+                        }
+                    }
+                }
+
+                updates.push_back(canonical_mutation(topology_mutation_builder(guard.write_timestamp())
+                                                             .set_transition_state(topology::transition_state::tablet_migration)
+                                                             .set_version(_topo_sm._topology.version + 1)
+                                                             .del_global_topology_request()
+                                                             .del_global_topology_request_id()
+                                                             .build()));
+                updates.push_back(canonical_mutation(topology_request_tracking_mutation_builder(utils::UUID{req_uuid})
+                                                             .done(error)
+                                                             .build()));
+                unsigned topo_muts_count = updates.size();
```

Comment:
why `unsigned`?
why define this variable at all?
updates.size() can be passed directly to the `mixed_change` ctor (and maybe validated there that it indeed fits into `unsigned int`).

Reply:
> why `unsigned`?

It can only hold a positive number.

> why define this variable at all? updates.size() can be passed directly to the `mixed_change` ctor (and maybe validated there that it indeed fits into `unsigned int`).

I have to save how many (topology) updates we inserted to the vector so far, because later I'm appending schema mutations to the same vector, as explained here https://github.com/scylladb/scylladb/pull/16723/commits/c7c2d940063fa9e34657a218cd02a4eac40a2d45#r1591562428

Reply:
> > why `unsigned`?
> 
> It can only hold a positive number.
> 

I meant, why not size_t?

> > why define this variable at all? updates.size() can be passed directly to the `mixed_change` ctor (and maybe validated there that it indeed fits into `unsigned int`).
> 
> I have to save how many (topology) updates we inserted to the vector so far, because later I'm appending schema mutations to the same vector, as explained here [c7c2d94#r1591562428](https://github.com/scylladb/scylladb/commit/c7c2d940063fa9e34657a218cd02a4eac40a2d45#r1591562428)

Hmm, I see.
Well, I agree with @kbr-scylla's comment: https://github.com/scylladb/scylladb/pull/16723/files#r1574577893 about just using 2 vectors in mixed_change and not try to optimize it with this count.

Reply:
The count has been removed and the mixed_change struct contains only 1 vector.

Reply:
The count has been removed and the mixed_change struct contains only 1 vector.